.circleci/config.yml

```diff
@@ -9,14 +9,35 @@ executors:
       CXXFLAGS: -DGEOS_INLINE
 
 commands:
-  install_macos_dependencies:
+  mac_deps:
     steps:
       - run: brew install protobuf cmake ccache libtool libspatialite pkg-config luajit curl wget czmq lz4 spatialite-tools unzip
-      - run: sudo python3 -m pip install shapely "conan<2.0.0"
+      - run: source venv/bin/activate && sudo python3 -m pip install shapely "conan<2.0.0"
       - run: git clone https://github.com/kevinkreiser/prime_server --recurse-submodules && cd prime_server && ./autogen.sh && ./configure && make -j8 && make install
-  install_linux_dependencies:
+  mac_preflight:
     steps:
-      - run: sudo bash ./scripts/install-linux-deps.sh
+      - run: python3 -m venv venv
+  linux_deps:
+    steps:
+      - run: |
+            source venv/bin/activate
+            ./scripts/install-linux-deps.sh
+  linux_preflight:
+    steps:
+      - run: |
+            sudo apt-get update --assume-yes
+            env DEBIAN_FRONTEND=noninteractive sudo apt-get install --yes --quiet python3-pip python3-venv git curl
+            python3 -m venv venv
```

Comment:
both mac and linux platforms now use venv to install python stuff. newer versions of python insist on this so rather than work around it (which you can) we just comply in this case